addons/mail/controllers/discuss.py

```diff
@@ -98,6 +98,8 @@ def _response_discuss_channel_invitation(self, channel_sudo, is_channel_token_se
                         'shouldAddGuestAsMemberOnJoin': True,
                         'shouldDisplayWelcomeViewInitially': True,
                     })
+                if not channel_sudo._can_invite_guest(guests=guest):
+                    raise NotFound()
```

Comment:
Because `_can_invite_guest` raises instead of returning a boolean, it's not a "if/raise" that is needed here but "try/except/raise"